src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -6,67 +6,55 @@
  * Side Public License, v 1.
  */
 
-import './search_embeddable.scss';
-import angular from 'angular';
-import _ from 'lodash';
 import { Subscription } from 'rxjs';
+import React from 'react';
+import ReactDOM from 'react-dom';
 import { i18n } from '@kbn/i18n';
-import { UiActionsStart } from '../../../../ui_actions/public';
-import { RequestAdapter, Adapters } from '../../../../inspector/public';
-import {
-  APPLY_FILTER_TRIGGER,
-  esFilters,
-  Filter,
-  TimeRange,
-  FilterManager,
-  Query,
-  IFieldType,
-} from '../../../../data/public';
+import { isEqual } from 'lodash';
 import { Container, Embeddable } from '../../../../embeddable/public';
-import * as columnActions from '../angular/doc_table/actions/columns';
-import searchTemplate from './search_template.html';
-import searchTemplateGrid from './search_template_datagrid.html';
 import { ISearchEmbeddable, SearchInput, SearchOutput } from './types';
-import { SortOrder } from '../angular/doc_table/components/table_header/helpers';
-import { getSortForSearchSource } from '../angular/doc_table';
-import { getServices, IndexPattern, ISearchSource } from '../../kibana_services';
+import { SavedSearch } from '../../saved_searches';
+import { Adapters, RequestAdapter } from '../../../../inspector/common';
+import { ISearchSource } from '../../../../data/common';
 import { SEARCH_EMBEDDABLE_TYPE } from './constants';
-import { SavedSearch } from '../..';
+import { APPLY_FILTER_TRIGGER, esFilters, FilterManager } from '../../../../data/public';
+import { DiscoverServices } from '../../build_services';
+import { Query, TimeRange } from '../../../../data/common/query';
+import { Filter } from '../../../../data/common/es_query/filters';
+import { SortOrder } from '../angular/doc_table/components/table_header/helpers';
+import { IFieldType } from '../../../../data/common/index_patterns/fields';
+import { ElasticSearchHit } from '../doc_views/doc_views_types';
+import { IndexPattern } from '../../../../data/common/index_patterns/index_patterns';
```

Comment:
there are plenty of imports from `data` which could be shortened, merged, also wonder why this is not throwing an error, since we shouldn't import from folders below `common`, `public`, `server`?

Reply:
good catch, and a good question 😬